### PR TITLE
#19 Github Workflows

### LINKED ISSUE #19: Dev Ops - Add Github Workflows
### Description

We want workflows to automate compilation so that github catches any errors for us/ makes sure that our files are formatted and theres no linting errors

### Acceptance Criteria

- [ ] Docker compose build succeeds
- [ ] Prettier Succeeds
- [ ] Linting Succeeds

### Proposed Solution

Github workflows (steal from finishline)

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Added Some Workflows for docker-compose build, prettier and linting 

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #19  (issue #)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,10 +2,25 @@
   "name": "telemetry-hub",
```
Comment (RChandler234) [inline package.json:2]: you should rename the project

Comment (RChandler234) [reply package.json:2]: also the scylla-server name in its package.json is wrong and should be updated to scylla (that's my b)

File: .github/workflows/build-check.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Compose
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install docker-compose
+
+      - name: Build Docker Compose
+        run: |
+          if ! docker-compose build; then
```
Comment (RChandler234) [inline .github/workflows/build-check.yml:25]: how long does this take to run usually out of curiosity?

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+module.exports = {
```
Comment (RChandler234) [inline .eslintrc.js:1]: you should have ppl install this plugin, it should let them know in their ide if there's an eslint error (might need to be configured a lil tho)
image

**Review by RChandler234 — CHANGES_REQUESTED** [review]
This all looks very lit, good shit to get setup before devs start wreaking havoc on the codebase 🔥 

When I npm run install-dependencies it creates a package-lock.json for the backend and root folder, you should p commit those to github as well

Also for Argos PRs, I'ma leave it up to your discretion when you want to merge them, I'm just leaving my comments to give input/suggestions

File: .github/workflows/build-check.yml (hunk 2 above)
Comment (Peyton-McKee) [reply .github/workflows/build-check.yml:25]: like 2 mins ish

**Pull request merged**